docfx/docs/cursor.md

```diff
@@ -0,0 +1,162 @@
+    - Backed with `private int _cursorStyle`
+    - On set, calls `OnCursorStyleChanged()`
+  * Add `internal abstract void OnCursorStyleChanged()`
+    - Called by `base` whenever the cursor style changes, but ONLY if `value != _cursorStyle`.
+
+  * Add `internal virtual (int Id, string StyleName) []  GetCursorStyles()`
+    - Returns an array of styles supported by the driver, NOT including Invisible. 
+    - The first item in array is always "Default".
+    - Base implementation returns `{ 0, "Default" }`
+    - `CursesDriver` and `WindowsDriver` will need to implement overrides.
+
+  * Add `internal Point? CursorPosition {get; internal set; }`
+    - Backed with `private Point? _cursorPosition`
+    - If `!HasValue` the cursor is not visible
+    - If `HasValue` the cursor is visible at the Point.
+    - On set, calls `OnCursorPositionChanged` ONLY if `value != _cursorPosition`.
+  * Add `internal abstract void OnCursorPositionChanged()`
+    - Called by `base` whenever the cursor position changes. 
+    - Depending on the value of `CursorPosition`:
+        - If `!HasValue` the cursor is not visible - does whatever is needed to make the cursor invisible.
+        - If `HasValue` the cursor is visible at the `CursorPosition` - does whatever is needed to make the cursor visible (using `CursorStyle`).
+
+  * Make sure the drivers only make the cursor visible (or leave it visible) when `CursorPosition` changes!
+
+### `Application`
+
+* 
+
+* Add `internal static View FocusedView {get; private set;}` 
+  - Backed by `private static _focusedView`
+  - On set, 
+    - if `value != _focusedView` 
+        - Unsubscribe from `_focusedView.CursorPositionChanged`
+        - Subscribe to `value.CursorPositionChanged += CursorPositionChanged`        
+        - `_focusedView = value`
+        - Call `UpdateCursor` 
+
+* Add `internal bool CursorPositionChanged (object sender, LocationChangedEventArgs a)`
+
+    Called when:
+
+    - `FocusedView`
+        - Has changed to another View (should cover `FocusedView.Visible/Enable` changes)
+        - Has changed layout - 
+        - Has changeed it's `CursorPosition`
+    - `CursorStyle` has changed
+
+    Does:
+
+    - If `FocusedView is {}` and `FocusedView.CursorPosition` is visible (e.g. w/in `FocusedView.SuperView.Viewport`) 
+        - Does `Driver.CursorPosition = ToScreen(FocusedView.CursorPosition)`
+    - Else
+        - Makes driver cursor invisible with `Driver.CursorPosition = null`
+
+* Add `public static int CursorStyle {get; internal set; }`
+  - Backed with `private static int _cursorStyle
+  - If `value != _cursorStyle`
+    - Calls `ConsoleDriver.CursorStyle = _cursorStyle` 
+    - Calls `UpdateCursor`
+
+* Add `public (int Id, string StyleName) []  GetCursorStyles()`
+  - Calls through to `ConsoleDriver.GetCursorStyles()`
+
+
+
+# Issues with Current Design
+
+## `Driver.Row/Pos`, which are changed via `Move` serves two purposes that confuse each other:
```

Comment:
I think you mean `Driver.Row/Col` because `Pos` can be referred to the `Row` or `Col`.